<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.9.1" xml:lang="en-US">
  <compounddef id="classNeoFOAM_1_1Error" kind="class" language="C++" prot="public">
    <compoundname>NeoFOAM::Error</compoundname>
    <basecompoundref prot="public" virt="non-virtual">std::exception</basecompoundref>
    <derivedcompoundref refid="classNeoFOAM_1_1DimensionMismatch" prot="public" virt="non-virtual">NeoFOAM::DimensionMismatch</derivedcompoundref>
    <includes refid="exceptions_8hpp" local="no">exceptions.hpp</includes>
      <sectiondef kind="private-attrib">
      <memberdef kind="variable" id="classNeoFOAM_1_1Error_1a9d61a74cf45457e4d043f41ba5ab9983" prot="private" static="no" mutable="no">
        <type>const std::string</type>
        <definition>const std::string NeoFOAM::Error::what_</definition>
        <argsstring></argsstring>
        <name>what_</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="/home/runner/work/NeoFOAM/NeoFOAM/include/NeoFOAM/helpers/exceptions.hpp" line="46" column="23" bodyfile="/home/runner/work/NeoFOAM/NeoFOAM/include/NeoFOAM/helpers/exceptions.hpp" bodystart="46" bodyend="-1"/>
      </memberdef>
      </sectiondef>
      <sectiondef kind="public-func">
      <memberdef kind="function" id="classNeoFOAM_1_1Error_1a9df5b85d215b4e57ef3240e3500773c7" prot="public" static="no" const="no" explicit="no" inline="yes" virt="non-virtual">
        <type></type>
        <definition>NeoFOAM::Error::Error</definition>
        <argsstring>(const std::string &amp;file, int line, const std::string &amp;what)</argsstring>
        <name>Error</name>
        <param>
          <type>const std::string &amp;</type>
          <declname>file</declname>
        </param>
        <param>
          <type>int</type>
          <declname>line</declname>
        </param>
        <param>
          <type>const std::string &amp;</type>
          <declname>what</declname>
        </param>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
<para>Initializes an error.</para>
<para><parameterlist kind="param"><parameteritem>
<parameternamelist>
<parametername>file</parametername>
</parameternamelist>
<parameterdescription>
<para>The name of the offending source file </para>
</parameterdescription>
</parameteritem>
<parameteritem>
<parameternamelist>
<parametername>line</parametername>
</parameternamelist>
<parameterdescription>
<para>The source code line number where the error occurred </para>
</parameterdescription>
</parameteritem>
<parameteritem>
<parameternamelist>
<parametername>what</parametername>
</parameternamelist>
<parameterdescription>
<para>The error message </para>
</parameterdescription>
</parameteritem>
</parameterlist>
</para>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="/home/runner/work/NeoFOAM/NeoFOAM/include/NeoFOAM/helpers/exceptions.hpp" line="34" column="5" bodyfile="/home/runner/work/NeoFOAM/NeoFOAM/include/NeoFOAM/helpers/exceptions.hpp" bodystart="34" bodyend="36"/>
      </memberdef>
      <memberdef kind="function" id="classNeoFOAM_1_1Error_1af83ca1338f3a80e3801a84d37351c924" prot="public" static="no" const="yes" explicit="no" inline="yes" noexcept="yes" virt="virtual">
        <type>const char *</type>
        <definition>virtual const char* NeoFOAM::Error::what</definition>
        <argsstring>() const noexcept override</argsstring>
        <name>what</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
<para>Returns a human-readable string with a more detailed description of the error. </para>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="/home/runner/work/NeoFOAM/NeoFOAM/include/NeoFOAM/helpers/exceptions.hpp" line="42" column="24" bodyfile="/home/runner/work/NeoFOAM/NeoFOAM/include/NeoFOAM/helpers/exceptions.hpp" bodystart="42" bodyend="42"/>
      </memberdef>
      </sectiondef>
    <briefdescription>
<para>Base class for consistent error representation. </para>
    </briefdescription>
    <detaileddescription>
<para>The <ref refid="classNeoFOAM_1_1Error" kindref="compound">Error</ref> class is used to report exceptional behaviour in library functions. <ref refid="namespaceNeoFOAM" kindref="compound">NeoFOAM</ref> uses C++ exception mechanism to this end, and the <ref refid="classNeoFOAM_1_1Error" kindref="compound">Error</ref> class represents a base class for all types of errors. The exact list of errors which could occur during the execution of a certain library routine is provided in the documentation of that routine, along with a short description of the situation when that error can occur. During runtime, these errors can be detected by using standard C++ try-catch blocks, and a human-readable error description can be obtained by calling the <ref refid="classNeoFOAM_1_1Error_1af83ca1338f3a80e3801a84d37351c924" kindref="member">Error::what()</ref> method. </para>
    </detaileddescription>
    <inheritancegraph>
      <node id="3">
        <label>NeoFOAM::DimensionMismatch</label>
        <link refid="classNeoFOAM_1_1DimensionMismatch"/>
        <childnode refid="1" relation="public-inheritance">
        </childnode>
      </node>
      <node id="1">
        <label>NeoFOAM::Error</label>
        <link refid="classNeoFOAM_1_1Error"/>
        <childnode refid="2" relation="public-inheritance">
        </childnode>
      </node>
      <node id="2">
        <label>std::exception</label>
      </node>
    </inheritancegraph>
    <collaborationgraph>
      <node id="1">
        <label>NeoFOAM::Error</label>
        <link refid="classNeoFOAM_1_1Error"/>
        <childnode refid="2" relation="public-inheritance">
        </childnode>
      </node>
      <node id="2">
        <label>std::exception</label>
      </node>
    </collaborationgraph>
    <location file="/home/runner/work/NeoFOAM/NeoFOAM/include/NeoFOAM/helpers/exceptions.hpp" line="23" column="1" bodyfile="/home/runner/work/NeoFOAM/NeoFOAM/include/NeoFOAM/helpers/exceptions.hpp" bodystart="24" bodyend="47"/>
    <listofallmembers>
      <member refid="classNeoFOAM_1_1Error_1a9df5b85d215b4e57ef3240e3500773c7" prot="public" virt="non-virtual"><scope>NeoFOAM::Error</scope><name>Error</name></member>
      <member refid="classNeoFOAM_1_1Error_1af83ca1338f3a80e3801a84d37351c924" prot="public" virt="virtual"><scope>NeoFOAM::Error</scope><name>what</name></member>
      <member refid="classNeoFOAM_1_1Error_1a9d61a74cf45457e4d043f41ba5ab9983" prot="private" virt="non-virtual"><scope>NeoFOAM::Error</scope><name>what_</name></member>
    </listofallmembers>
  </compounddef>
</doxygen>
